src/display/metadata.js

```diff
@@ -13,43 +13,50 @@
  * limitations under the License.
  */
 
-function fixMetadata(meta) {
-  return meta.replace(/>\\376\\377([^<]+)/g, function(all, codes) {
-    var bytes = codes.replace(/\\([0-3])([0-7])([0-7])/g,
-                              function(code, d1, d2, d3) {
-      return String.fromCharCode(d1 * 64 + d2 * 8 + d3 * 1);
-    });
-    var chars = '';
-    for (var i = 0; i < bytes.length; i += 2) {
-      var code = bytes.charCodeAt(i) * 256 + bytes.charCodeAt(i + 1);
-      chars += (code >= 32 && code < 127 && code !== 60 && code !== 62 &&
-        code !== 38) ? String.fromCharCode(code) :
-        '&#x' + (0x10000 + code).toString(16).substring(1) + ';';
+import { SimpleXMLParser } from './dom_utils';
+
+class Metadata {
+  constructor(data) {
+    if (typeof data === 'string') {
+      // Ghostscript may produce invalid metadata, so try to repair that first.
+      data = this._repair(data);
+
+      // Convert the string to a DOM `Document`.
+      let parser = new SimpleXMLParser();
+      data = parser.parseFromString(data);
+    } else if (!(data instanceof Document)) {
+      throw new Error('Metadata: input is not a string or `Document`');
```

Comment:
do we need `instanceof Document` check? if not let's replace `if (typeof data === 'string'` with assert.